File: scripts/algorithms/alternative-implementations/pagerank/py/pagerank.py
from scipy.io import mmread
from scipy.sparse import coo_array
import time
import sys
import numpy as np

def pagerank(filename, maxi=250):
    G = mmread(filename)
    start = time.time()
    n = G.shape[0]
    p = np.ones(n)
    alpha = 0.85
    one_minus_alpha = 1 - alpha

    for iter in range(maxi):
        p = G.dot(p) * alpha + p * one_minus_alpha 
        p = p / p.sum()
    end = time.time()
    print(end - start)

if __name__ == "__main__":
    args = sys.argv
    assert len(args) == 2
    filename = args[1]
    pagerank(filename)

File: scripts/algorithms/alternative-implementations/pagerank/jl/pagerank.jl
using MatrixMarket

function pagerank(filename, maxi)
  G = MatrixMarket.mmread(filename)
  n = size(G, 1)
  p = ones(n)
  alpha = 0.85
  one_minus_alpha = 1 - alpha

  start = time_ns()
  for iter in 1:maxi
    p = (G * p) * alpha + p * one_minus_alpha
    p = p / sum(p)
  end
  fin = time_ns()
  println((fin - start) * 1e-9)
end

@assert(length(ARGS) == 1)
filename = ARGS[1]
maxi = 250
pagerank(filename, maxi)

